extension/bootstrap.js

```diff
@@ -44,12 +44,37 @@ const REASONS = {
 };
 const UI_AVAILABLE_NOTIFICATION = "sessionstore-windows-restored";
 const PANEL_CSS_URI = Services.io.newURI('resource://pioneer-study-online-news/content/panel.css');
-
+const EXPIRATION_DATE_PREF = "extensions.pioneer-online-news.expirationDateString";
 
 this.Bootstrap = {
   install() {},
 
-  startup(data, reason) {
+  async startup(data, reason) {
+    // Check if the user is opted in to pioneer and if not end the study
+    Pioneer.startup();
+
+    const isEligible = await Pioneer.utils.isUserOptedIn();
+    if (!isEligible) {
+      Pioneer.utils.endStudy("ineligible");
+      return;
+    }
+
+    // Always set EXPIRATION_DATE_PREF if it not set, even if outside of install.
+    // This is a failsafe if opt-out expiration doesn't work, so should be resilient.
+    if (!Services.prefs.prefHasUserValue(EXPIRATION_DATE_PREF)) {
+      const now = new Date(Date.now());
+      // 70, below, is the number of days in 10 weeks.
+      const expirationDateString = new Date(now.setDate(now.getDate() + 70)).toISOString();
```

Comment:
I don't think this really matters that much.   Adding in the extra code to extract the 10 week value from the Config object means we have to add more code here to convert the ms duration of 10 weeks back to number of days.

Reply:
Err... on second thought.  We probably want to just use ms timestamps everywhere instead of ISO dates after looking at the date comparison code below.